Apply latest suggested project settings

webframp opened the pull request:
As suggested by Xcode 11.4.1, using defaults

webframp committed:
Apply latest suggested project settings

As suggested by Xcode 11.4.1, using defaults

webframp commented:
Just posting this before I go to sleep, so I remember to come back to it.

FYI @dive only two warnings currently, both in https://github.com/MobileOrg/mobileorg/blob/develop/Classes/Utilities/UIAlertController%2BMobileOrg.swift#L47

`Cast from 'UIViewController?.Type' to unrelated type 'UINavigationController.Type' always fails`

Will try to find time this week to resolve.

dive commented on MobileOrg.xcodeproj/project.pbxproj:
```diff
@@ -1086,6 +1112,32 @@
 		79BA55701073A82C00D67917 /* AppStore */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
+				CLANG_WARN_BOOL_CONVERSION = YES;
+				CLANG_WARN_COMMA = YES;
+				CLANG_WARN_CONSTANT_CONVERSION = YES;
+				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
+				CLANG_WARN_EMPTY_BODY = YES;
+				CLANG_WARN_ENUM_CONVERSION = YES;
+				CLANG_WARN_INFINITE_RECURSION = YES;
+				CLANG_WARN_INT_CONVERSION = YES;
+				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
+				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
+				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
+				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
+				CLANG_WARN_STRICT_PROTOTYPES = YES;
+				CLANG_WARN_SUSPICIOUS_MOVE = YES;
+				CLANG_WARN_UNREACHABLE_CODE = YES;
+				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
```
The same as above.

webframp replied:
Removed

dive commented on MobileOrg.xcodeproj/project.pbxproj:
```diff
@@ -1073,6 +1073,32 @@
 		79BA556E1073A82500D67917 /* AdHoc */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
+				CLANG_WARN_BOOL_CONVERSION = YES;
+				CLANG_WARN_COMMA = YES;
+				CLANG_WARN_CONSTANT_CONVERSION = YES;
+				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
+				CLANG_WARN_EMPTY_BODY = YES;
+				CLANG_WARN_ENUM_CONVERSION = YES;
+				CLANG_WARN_INFINITE_RECURSION = YES;
+				CLANG_WARN_INT_CONVERSION = YES;
+				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
+				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
+				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
+				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
+				CLANG_WARN_STRICT_PROTOTYPES = YES;
+				CLANG_WARN_SUSPICIOUS_MOVE = YES;
+				CLANG_WARN_UNREACHABLE_CODE = YES;
+				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
```
We already have all these warnings enabled in the `Global.shared.xcconfig`. It seems that Xcode just ignores the `.xconfig` when suggesting such changes. I think we can revert them, bump for the `LastUpgradeCheck` will resolve this warning from Xcode.

webframp replied:
Makes sense to me. A bit annoying that it ignores the `.xconfig` in this case

webframp replied:
Good suggestion, bumped check version in ea63d7165bb0eee51802497f007bb7d71f30943e

Always hate dealing with project files "managed" by Xcode

dive commented on MobileOrg.xcodeproj/project.pbxproj:
```diff
@@ -1099,6 +1151,34 @@
 		C01FCF4F08A954540054247B /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
+				CLANG_ANALYZER_LOCALIZABILITY_NONLOCALIZED = YES;
+				CLANG_WARN_BLOCK_CAPTURE_AUTORELEASING = YES;
+				CLANG_WARN_BOOL_CONVERSION = YES;
+				CLANG_WARN_COMMA = YES;
+				CLANG_WARN_CONSTANT_CONVERSION = YES;
+				CLANG_WARN_DEPRECATED_OBJC_IMPLEMENTATIONS = YES;
+				CLANG_WARN_EMPTY_BODY = YES;
+				CLANG_WARN_ENUM_CONVERSION = YES;
+				CLANG_WARN_INFINITE_RECURSION = YES;
+				CLANG_WARN_INT_CONVERSION = YES;
+				CLANG_WARN_NON_LITERAL_NULL_CONVERSION = YES;
+				CLANG_WARN_OBJC_IMPLICIT_RETAIN_SELF = YES;
+				CLANG_WARN_OBJC_LITERAL_CONVERSION = YES;
+				CLANG_WARN_RANGE_LOOP_ANALYSIS = YES;
+				CLANG_WARN_STRICT_PROTOTYPES = YES;
+				CLANG_WARN_SUSPICIOUS_MOVE = YES;
+				CLANG_WARN_UNREACHABLE_CODE = YES;
+				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
+				ENABLE_STRICT_OBJC_MSGSEND = YES;
+				ENABLE_TESTABILITY = YES;
+				GCC_NO_COMMON_BLOCKS = YES;
+				GCC_WARN_64_TO_32_BIT_CONVERSION = YES;
+				GCC_WARN_ABOUT_RETURN_TYPE = YES;
+				GCC_WARN_UNDECLARED_SELECTOR = YES;
+				GCC_WARN_UNINITIALIZED_AUTOS = YES;
+				GCC_WARN_UNUSED_FUNCTION = YES;
+				GCC_WARN_UNUSED_VARIABLE = YES;
+				ONLY_ACTIVE_ARCH = YES;
```
The same as above.

webframp replied:
Removed

dive reviewed (commented):
Thanks! 

Yeah, it seems that we have to rewrite the `mo_presentingViewController` to avoid these "Cast always fails" warnings from the compiler. I suggest something like this:

```swift
private static var mo_presentingViewController: UIViewController? {
    let rootViewController = UIApplication.shared.delegate?.window??.rootViewController
    if let controller = (rootViewController as? UINavigationController)?.topViewController {
        return controller
    } else if let controller = (rootViewController as? UITabBarController)?.selectedViewController {
        return controller
    }
    return rootViewController
}
```

webframp committed:
Revert warnings config, bump LastUpgradeCheck 1140

Since we already have these defined in `Global.shared.xcconfig` as
discussed in #262 revert the automated changes applied by Xcode and bump
the upgrade check value to bypass.

webframp committed:
Refactor mo_presentingViewController

This is specifically to avoid recent "Cast always fails" compiler
warnings. Tested a few minor different approaches to this refactor but
just decided to go with what was suggested by @dive on #262 since it
seems to work well for me in testing.

https://github.com/MobileOrg/mobileorg/pull/262#pullrequestreview-408652708

webframp commented:
@dive updated based on feedback, thanks. I think this is good to go if you can take one last look.

webframp commented:
Tested this a bit more, seems fine and warnings are gone after last 2 commits here. Going to merge.
